cmd/admin-profiling-stop.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * Minio Client (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/mc/pkg/probe"
+)
+
+var adminProfilingStopFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "save",
+		Usage: "The location of the zipped profiling files",
+	},
+}
+
+var adminProfilingStopCmd = cli.Command{
+	Name:            "stop",
+	Usage:           "Stop and download profiling data",
+	Action:          mainAdminProfilingStop,
+	Before:          setGlobalsFromContext,
+	Flags:           append(adminProfilingStopFlags, globalFlags...),
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] TARGET
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+    2. Download latest profiling data under the specified filename
+       $ {{.HelpName}} --save /tmp/profiling-data.zip myminio/
+
+`,
+}
+
+func checkAdminProfilingStopSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 1 {
+		cli.ShowCommandHelpAndExit(ctx, "stop", 1) // last argument is exit code
+	}
+
+	if len(ctx.String("save")) == 0 {
+		fatalIf(errDummy(), "Kindly specify the save path of profiling data using --save flag.")
```

Comment:
why not use current directory as default? and keep the name of the zip file same as the profiler type?

Reply:
> why not use current directory as default? and keep the name of the zip file same as the profiler type?

Umm, mc start doesn't keep stateful data in the disk so mc stop will remember what type of profiling this is about, but we can update profiling adming API to return the type of profiling data I guess, not sure if this is worth to do

Reply:
> No, not that I meant simply say `profiling.zip` @vadmeste if the user is expected to know what profile type it is - then it doesn't matter what we choose. We should do this automatically I mean.

Yes, I see now, it makes sense, fixed now